src/backend/catalog/aclchk.c

```diff
@@ -2324,14 +2325,14 @@ CopyRelationAcls(Oid srcId, Oid destId)
 		if (attSrcForm->attisdropped)
 			continue;
 
+		attDestTuple = &destAttList->members[dest_i++]->tuple;
```

Comment:
this line also can be moved closer to using

Reply:
the problem is that we will then need to save the index `dest_i` somewhere since we need to increment it after `if (attSrcForm->attisdropped)` but before `if (isNull)`